### PR TITLE
fix: use context for config

### PR DESCRIPTION
**Pull request by sebastianquek**
The existing `useConfig` returns a stale config while it waits for the async get call to resolve each time `useConfig` is 'mounted'. By relying on context, we can ensure that the config obtained from AsyncStorage is resolved before any of the context consumers mount/render. Subsequent calls to get/set the config relies on the context and the updating of AsyncStorage becomes a side effect.

### CODE REVIEW

File: src/context/config.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      [key]: value
+    };
+    setConfig(nextConfig);
+    AsyncStorage.setItem(CONFIG_KEY, JSON.stringify(nextConfig));
+  };
+
+  const loadConfigFromStore = async (): Promise<void> => {
+    const configStr = await AsyncStorage.getItem(CONFIG_KEY);
+    if (configStr) {
+      setConfig(JSON.parse(configStr));
```
Comment (Nebulis) [inline src/context/config.tsx]: there is one problem with this, if you add a new settings and there already exist a stored config, then there will be no default value provided for the new setting

=> https://github.com/Open-Attestation/identity-wallet/pull/63/files#diff-5689e8dfaa1efe9470061c2814ecf3e0R37

File: src/context/config.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+const CONFIG_KEY = "CONFIG";
+const DEFAULT_CONFIG: Config = { appMode: AppMode.production };
+
+const ConfigContext = createContext<ConfigContext>({
+  config: DEFAULT_CONFIG,
+  setConfigValue: () => {} // eslint-disable-line @typescript-eslint/no-empty-function
+});
+
+export const useConfigContext = (): ConfigContext =>
```
Comment (Nebulis) [inline src/context/config.tsx]: you could have kept useConfig as a name

File: src/context/config.tsx (hunk 1 above)
Comment (sebastianquek) [reply src/context/config.tsx]: Ok will add this in

File: src/context/config.tsx (hunk 2 above)
Comment (sebastianquek) [reply src/context/config.tsx]: I left the `Context` in there to remind the consumer not use this without ensuring an enclosed context.

**Pull request merged**